torch/distributions/distribution.py

```diff
@@ -220,6 +241,19 @@ def _validate_sample(self, value):
         if not self.support.check(value).all():
             raise ValueError('The value argument must be within the support')
 
+    def _get_checked_instance(self, cls, instance=None):
+        def cons(cls_):
+            """
+            Returns the `__init__` function for a python 2.7 / 3 class.
+            """
+            return getattr(cls_.__init__, "__func__", cls_.__init__)
+
+        if instance is None and cons(type(self)) is not cons(cls):
```

Comment:
You can simplify to `type(self).__init__ == cls.__init__`, which appears to work in both Python 2.7 and 3.

Reply:
You are right! Equality works fine actually (they have the same id), but `is` fails.